crates/map/src/finders/dfs.cairo

```diff
@@ -0,0 +1,179 @@
+                    let neighbor = NodeTrait::new(neighbor_position, current.position, 0, 0);
+                    stack.append(neighbor);
+                }
+            };
+        };
+
+        // Reconstruct and return the path if found
+        if !path_found {
+            return array![].span();
+        };
+        Self::path(parents, start, target)
+    }
+
+    /// Reconstructs the path from start to target using the parents dictionary
+    #[inline]
+    fn path(mut parents: Felt252Dict<u8>, start: Node, target: Node) -> Span<u8> {
+        let mut path: Array<u8> = array![];
+        let mut current = target.position;
+
+        loop {
+            if current == start.position {
+                break;
+            }
+            path.append(current);
+            current = parents.get(current.into());
+        };
+
+        path.span()
+    }
+}
+#[cfg(test)]
+mod test {
+    // Local imports
+    use super::DFS;
+
+    #[test]
+    fn test_dfs_search_small() {
+        // x───┐
+        // 1 0 │
+        // 0 1 s
+        let grid: felt252 = 0x1EB;
+        let width = 3;
+        let height = 3;
+        let from = 0;
+        let to = 8;
+        let path = DFS::search(grid, width, height, from, to);
+        assert_eq!(path, array![8, 7, 6, 3].span());
+    }
+
+    #[test]
+    fn test_dfs_search_impossible() {
+        // x 1 0
+        // 1 0 1
+        // 0 1 s
+        let grid: felt252 = 0x1AB;
+        let width = 3;
+        let height = 3;
+        let from = 0;
+        let to = 8;
+        let path = DFS::search(grid, width, height, from, to);
+        assert_eq!(path, array![].span());
+    }
+
+    #[test]
+    fn test_dfs_search_medium() {
+        // ┌─x 0 0
+        // │ 0 1 1
+        // └─────┐
+        // 1 1 1 s
+        let grid: felt252 = 0xCBFF;
+        let width = 4;
+        let height = 4;
+        let from = 0;
+        let to = 14;
+        let path = DFS::search(grid, width, height, from, to);
+        assert!(
+            path.len() > 0
+        ); // DFS may not find the shortest path, so we just check if a path is found
+    }
+
+    #[test]
+    fn test_dfs_single_cell_path() {
+        // Grid representation:
+        // x s
+        // 1 1
+        let grid: felt252 = 0xF;
+        let width = 2;
+        let height = 2;
+        let from = 0;
+        let to = 1;
+        let path = DFS::search(grid, width, height, from, to);
+        assert_eq!(path, array![1].span());
+    }
+
+    #[test]
+    fn test_dfs_maze() {
+        // Grid representation:
+        // x 1 0 0 0
+        // 0 1 1 1 0
+        // 0 0 0 1 0
+        // 1 1 1 1 s
+        let grid: felt252 = 0xC385F;
+        let width = 5;
+        let height = 4;
+        let from = 0;
+        let to = 19;
+        let path = DFS::search(grid, width, height, from, to);
+        assert!(
+            path.len() > 0
+        ); // DFS may not find the shortest path, so we just check if a path is found
```

Comment:
There is only on path for this grid, so it should find the right one.
You should verify the path even if it is not the shortest to ensure non-regression.

Reply:
Hey since its DFS can only check if the path is a valid one or not since it doesnt guarantee one path ?